encryption.html

```diff
@@ -0,0 +1,237 @@
+          BLE advertisement, is composed as follows.
+        </p>
+        <ul>
+          <li>UUID <code>d2fc</code></li>
+          <li>BTHome device data byte <code>42</code></li>
+          <li>Ciphertext <code>696c8c91856f</code></li>
+          <li>Counter <code>00112233</code></li>
+          <li>Message Integrity Check (MIC) <code>045bbc2a</code></li>
+        </ul>
+        <p>
+          Service Data = <code>d2fc41696c8c91856f00112233045bbc2a</code>
+        </p>
+
+        <h3 id="decryption">Decrypting your messages</h3>
+        <p>
+          So, now we have encrypted our first message. But how to decrypt it?
+          Let's assume you received the above service data
+          <code>d2fc41696c8c91856f00112233045bbc2a</code>. Your sensor device
+          will also broadcast its MAC address in the header. The only thing you
+          need is the encryption key, which you should have written down when
+          you encrypted the message.
+        </p>
+        <p>
+          First, break down the service data into the following parts.
+        </p>
+          <ul>
+            <li>UUID: First 2 bytes, should be <code>d2fc</code></li>
+            <li>BTHome device data: 3rd byte, should be <code>41</code></li>
+            <li>Ciphertext: 4th byte from the start till 8th byte from the end:
+              <code>696c8c91856f</code>
+            </li>
+            <li>Counter: 8th byte from the end till 4th byte from the end:
+              <code>00112233</code>
+            </li>
+            <li>Message Integrity Check (MIC): 8th byte from the end till 4th byte from the end:
+              <code>045bbc2a</code>
+            </li>
+          </ul>
+        <p>
+          With this information, we can recreate the <strong>nonce</strong> again.
+        </p>
+          <ul>
+            <li>MAC address <code>5448E68F80A5</code></li>
+            <li>UUID <code>d2fc</code></li>
+            <li>BTHome device data byte <code>42</code></li>
+            <li>Counter <code>00112233</code></li>
+          </ul>
+        <p>
+          nonce = <code>5448e68f80a5d2fc4200112233</code>
+        </p>
+        <p>
+          With the Nonce, the MIC and the encryption key, we can decode the data
+          with the following commands.
+        </p>
+        <code>
+          from Cryptodome.Cipher import AES<br /><br />
+          cipher = AES.new(key, AES.MODE_CCM, nonce=nonce, mac_len=4)<br />
+          cipher.update(b"\x11")<br />
+          decrypted_data = cipher.decrypt_and_verify(ciphertext, mic)<br />
+        </code>
+        <p>
+          The result will be the decrypted data <code>02ca09 03bf13</code> like
+          we had on the top of this page.
+        </p>
+        <h3 id="examples">Example script</h3>
+        <p>
+          A detailed
+          <a
+            href="https://github.com/Bluetooth-Devices/bthome-ble/blob/main/src/bthome_ble/bthome_v2_encryption.py"
```

Comment:
Dead link?

Reply:
The links should be correct after the release.